File: apps/web/src/lib/queries/chunks.ts
import { useQuery } from "@tanstack/react-query";

import { apiGet } from "@/lib/api";

export type ChunkDetail = {
  chunk_id: string;
  meeting_id: string;
  document_id: string;
  chunk_index: number;
  text: string;
  document_filename: string | null;
  document_original_filename: string | null;
  document_doc_type: string | null;
};

export function useChunkDetail(chunkId: string | null) {
  return useQuery({
    queryKey: ["chunk-detail", chunkId],
    queryFn: () => apiGet<ChunkDetail>(`/chunks/${chunkId}`),
    enabled: Boolean(chunkId),
  });
}

File: apps/api/app/schemas/chunks.py
from pydantic import BaseModel


class ChunkDetailResponse(BaseModel):
    chunk_id: str
    meeting_id: str
    document_id: str
    chunk_index: int
    text: str
    document_filename: str | None = None
    document_original_filename: str | None = None
    document_doc_type: str | None = None
